BaseTools/Plugin/HostBasedUnitTestRunner/HostBasedUnitTestRunner.py

```diff
@@ -207,14 +207,14 @@ def gen_code_coverage_msvc(self, thebuilder):
         # Generate coverage file
         coverageFile = ""
         for testFile in testList:
-            ret = RunCmd("OpenCppCoverage", f"--source {workspace} --export_type binary:{testFile}.cov -- {testFile}")
+            ret = RunCmd("OpenCppCoverage", f"--source {workspace} --export_type binary:{testFile}.cov -- {testFile}", workingdir=f"{workspace}Build/")
             coverageFile += " --input_coverage=" + testFile + ".cov"
             if ret != 0:
                 logging.error("UnitTest Coverage: Failed to collect coverage data.")
                 return 1
 
         # Generate and XML file if requested.by each package
-        ret = RunCmd("OpenCppCoverage", f"--export_type cobertura:{os.path.join(buildOutputBase, 'coverage.xml')} --working_dir={workspace}Build {coverageFile}")

```

Comment:
Instead of doing the complicated mess of checking if workspace ends with the os path separator (L206), you could use the pathlib library which will handle that for you. `workspace = Path(builder.env.GetValue("WORKSPACE"))`. Then you can do things like `workingdir=str(workspace / "Build")`

Pathlib could be used instead of os.path.join in many places here.

Up to you if you want to make those changes or not. It technically works, so :shrug:

Reply:
Yeah, this one came from upstream. I'm not sure if I care too much one way or the other. I'm happy to throw in the change though.